### DISCUSSION TITLE
Calling C++ from Python

### DISCUSSION THREAD

**Discussion by zhejiangzhoulei** [ds4dm/ml4co-competition#22] [Q&A] [answered]
Hi,
We are wondering if we are allowed to call c++ from python in our agent. 

Best

**Comment by gasse** [comment, marked as answer]
Hi @zhejiangzhoulei ,

Sure, you can call any library / program written any language, as long as your agent follows our Python API and does not use any commercial software (see the [rules](https://www.ecole.ai/2021/ml4co-competition/#rules)).

Also, mind that we must be able to install any software you use on our side as well for evaluation. It should be possible for us to compile C++ code if need be, but please make sure that the installation process is clear and simple (ideally a `pip install url` command), and also make sure that you manage to install your software stack within the singularity container we provide (see instructions to do so [here](PIPELINE.md)), before you send us your code.

Best,
Maxime

**Reply by zhejiangzhoulei** [reply]
hi, @gasse , thank you for your response :)
